app/controllers/avalon_marker_controller.rb

```diff
@@ -0,0 +1,135 @@
+# Copyright 2011-2015, The Trustees of Indiana University and Northwestern
+#   University.  Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+#   under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+#   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+#   specific language governing permissions and limitations under the License.
+# ---  END LICENSE_HEADER BLOCK  ---
+
+# Controller class for the AvalonMarker Model
+# Implements show, create, update, and delete
+class AvalonMarkerController < ApplicationController
+  # after_action:
+  def initialize
+    @not_found_messages = { marker: 'Marker Not Found',
+                            master_file: 'Master File Not Found',
+                            playlist_item: 'Playlist Item Not Found' }
+    @attr_keys = [:title, :start_time]
+  end
+
+  # Finds the marker based on passed uuid and renders the marker's json
+  # @param [Hash] params the parameters used by the controller
+  # @option params [String] :id the uuid of the marker you wish to render
+  # @example Rails Console Call To Show Marker that is resolved by AvalonMarker.where(uuid: '56')[0]
+  #    app.get('/avalon_marker/56')
+  def show
+    lookup_marker
+    render json: @marker.pretty_annotation
+  end
+
+  # Creates a marker and renders it as JSON
+  # @param [Hash] params the parameters used for creating a marker
+  # @option params [String] :master_file the pid of the MasterFile to be used for this marker's source
+  # @option params [String] :playlist_item the pid of the PlaylistItem to which this marker belongs
+  # @option params [String] :title the title to use for the marker, defaults the the MasterFile title
+  # @option params [String] :start_time the time point the marker begins, defaults to 0
+  # @example Rails Console command to create a marker based of the MasterFile whose pid is avalon:20 and default values for all other fields in the marker
+  #    app.post('/avalon_marker/', {master_file: 'avalon:20'})
+  def create
+    marker_params = params['avalon_marker']
+    render json: { message: 'Masterfile Not Supplied' }, status: 400 and return if marker_params[:master_file].nil?
+    render json: { message: 'Playlist Item Not Supplied' }, status: 400 and return if marker_params[:playlist_item].nil?
+    mf = MasterFile.where(id: marker_params[:master_file])[0]
+    render json: { message: @not_found_messages[:master_file] }, status: 400 and return if mf.nil?
+    pi = PlaylistItem.where(id: marker_params[:playlist_item])[0]
+    render json: { message: @not_found_messages[:playlist_item] }, status: 400 and return if pi.nil?
+
+    @marker = AvalonMarker.create(master_file: mf, playlist_item: pi)
```

Comment:
Can you use build instead of create here so the marker only has to be saved once?  Also I think you could also be rendering unsaved values if your persisted check passes because the create succeeded but the selected_key_updates failed.